include/aspect/geometry_model/spherical_shell.h

```diff
@@ -147,23 +147,23 @@ namespace aspect
           virtual Point<dim>
           get_new_point(const ArrayView<const Point<dim>> &vertices,
                         const ArrayView<const double>          &weights) const override;
+          /**
+           * Return the topography of the surface directly above the point given
+           * by the coordinates stored in the argument.
+           */
+          double topography_for_point (const Point<dim> &x_y_z) const;
 
-        private:
           /**
            * A pointer to the topography model.
            */
           const InitialTopographyModel::Interface<dim> *topo;
 
+        private:
```

Comment:
You are making `topo` a `public` variable. Let's try and avoid that -- I don't think you need it anyway, since you make the `topography_for_point()` function `public` already.